### PR TITLE
Overlay icon getting cached as well;  fixes #306

### LINKED ISSUE #306: Add overlay icons to IconCache
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4-IME
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Overlayed icons do not get cached currently, Might require small refactor to include overlay state to the cache.

### Logs

  - [ ] There are no logs regarding the issue
  - { ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- CDP4 version:
  - [ ] CDP4-IME: 
  - [ ] Other: 
  -
- Web Services:
  - [ ] CDP4 Web Services version: 
  - [ ] OCDT WSP version: 
- Environment (Operating system, version and so on):
  - [ ] Microsoft Excel (32/64 bit): 
  - [ ] Windows OS (32/64 bit): 
  - [ ] Additional information:

### PR DESCRIPTION
**Pull request by nathanatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
○ Implemented QueryOverlayBitmapSource method to actually cache overlay icon
○ Added a unit test;  fixes #306

### CODE REVIEW

File: CDP4Composition.Tests/IconCacheService/IconCacheServiceIconCachServiceTestFixture.cs [hunk 1]
```diff
@@ -61,5 +61,34 @@ public void VerityThatSameBitmapSourceIsReturned()
...
+            Assert.IsInstanceOf<BitmapSource>(firstBitmapSource);
+
+            stopWatchSecondFetch.Start();
+            var secondBitmapSource = this.iconCacheService.QueryOverlayBitmapSource(imageUri, overlayUri);
+            Assert.IsInstanceOf<BitmapSource>(secondBitmapSource);
+            stopWatchSecondFetch.Stop();
+
+            Assert.AreSame(firstBitmapSource, secondBitmapSource);
+
+            Assert.Greater(stopWatchFirstFetch.ElapsedMilliseconds, stopWatchSecondFetch.ElapsedMilliseconds);
```
Comment (lxatrhea) [inline CDP4Composition.Tests/IconCacheService/IconCacheServiceIconCachServiceTestFixture.cs]: When for some reason (PC is busy) the second time DOES take longer to execute, you get a false negative.
The Assert.AreSame(firstBitmapSource, secondBitmapSource) already makes sure the same object is returned.

File: CDP4Composition/Services/IconCacheService/IconCacheService.cs [hunk 2]
```diff
@@ -23,21 +23,26 @@ public class IconCacheService : IIconCacheService
...
+        private readonly Dictionary<Uri, BitmapImage> bitmapImages;
 
         /// <summary>
         /// The cache of error overlay <see cref="BitmapSource"/>s
         /// </summary>
-        private Dictionary<Uri, BitmapSource> bitmapSources;
+        private readonly Dictionary<Uri, BitmapSource> bitmapSources;
+
+        /// <summary>
+        /// The cache of any oher than error overlay <see cref="BitmapSource"/>s
```
Comment (lxatrhea) [inline CDP4Composition/Services/IconCacheService/IconCacheService.cs]: oher?

File: CDP4Composition.Tests/IconCacheService/IconCacheServiceIconCachServiceTestFixture.cs (hunk 1 above)
Comment (alexatrhea) [reply CDP4Composition.Tests/IconCacheService/IconCacheServiceIconCachServiceTestFixture.cs]: Indeed, here maybe better to assert that a certain method in `iconCacheService` is called?

File: CDP4Composition/Services/IconCacheService/IIconCacheService.cs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ﻿// --------------------------------------------------------------------------------------------------------------------
 // <copyright file="IIconCacheService.cs" company="RHEA System S.A.">
-//   Copyright (c) 2016 RHEA System S.A. All rights reserved
+//   Copyright (c) 2016-2020 RHEA System S.A. All rights reserved
```
Comment (alexatrhea) [inline CDP4Composition/Services/IconCacheService/IIconCacheService.cs]: No need to update files that you dont really touch. Its fine here, but keep that in mind.

File: CDP4Composition/Utilities/IconUtilities.cs [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ﻿// -----------------------------------------------------------------------------------------------
 // <copyright file="IconUtilities.cs" company="RHEA System S.A.">
-//   Copyright (c) 2015-2019 RHEA System S.A.
+//   Copyright (c) 2015-2020 RHEA System S.A.
```
Comment (alexatrhea) [inline CDP4Composition/Utilities/IconUtilities.cs]: Same as before. No need to commit files for year change.

**Pull request merged**